framework/src/fvkernels/FVDiffusion.C

```diff
@@ -48,15 +48,26 @@ FVDiffusion::FVDiffusion(const InputParameters & params)
 ADReal
 FVDiffusion::computeQpResidual()
 {
-  auto dudn = gradUDotNormal();
+  using namespace Moose::FV;
 
+  auto dudn = gradUDotNormal();
   ADReal coeff;
-  interpolate(_coeff_interp_method,
-              coeff,
-              _coeff(elemFromFace()),
-              _coeff(neighborFromFace()),
-              *_face_info,
-              true);
+
+  // If we are on internal faces, we interpolate the diffusivity as usual
+  if (_var.isInternalFace(*_face_info))
+    interpolate(_coeff_interp_method,
+                coeff,
+                _coeff(elemFromFace()),
+                _coeff(neighborFromFace()),
+                *_face_info,
+                true);
+  // Else we use linear inter/extrapolation using regular face args or single-sided
+  // face arguments
+  else
+  {
+    const auto face = singleSidedFaceArg();
+    coeff = _coeff(face);
+  }
```

Comment:
👍 for now. In the future, as we talked about, we will make it so kernel developers don't have to do this `if-else` branching, but rather we will handle it in the functors themselves based on the blocks of the consumer

Reply:
Roger that!